modules/storage_account/storage_account.tf

```
@@ -39,24 +39,19 @@ resource "azurerm_storage_account" "stg" {
   table_encryption_key_type         = try(var.storage_account.table_encryption_key_type, null)
   tags                              = merge(local.tags, try(var.storage_account.tags, null), local.caf_tags)
   public_network_access_enabled     = try(var.storage_account.public_network_access_enabled, null)
-
+  shared_access_key_enabled         = try(var.storage_account.shared_access_key_enabled, true)
+  default_to_oauth_authentication   = try(var.storage_account.default_to_oauth_authentication, false)
```

```suggestion
  default_to_oauth_authentication   = try(var.storage_account.default_to_oauth_authentication, null)
```
Prefer to defer to provider defaults